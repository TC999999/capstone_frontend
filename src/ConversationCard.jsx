import { Link } from "react-router-dom";

const ConversationCard = ({ message }) => {
  return (
    <div className="conversation-card">
      <p>
        <b>From:</b>{" "}
        <Link to={`/users/${message.from_username}`}>
          {message.from_username}
        </Link>
      </p>
      <p>
        <b>To:</b>{" "}
        <Link to={`/users/${message.to_username}`}>{message.to_username}</Link>
      </p>
      <p>
        <b>About:</b> {message.item_name}
      </p>
      <p>
        <b>Body:</b> {message.body}
      </p>
      <small>
        <i>sent at:</i> {message.sent_at}
      </small>
    </div>
  );
};

export default ConversationCard;
