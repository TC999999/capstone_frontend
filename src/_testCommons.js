const TEST_USER = {
  username: "u1",
  firstName: "fn1",
  lastName: "ln1",
  email: "testemail1@gmail.com",
  address: "100 address",
  zipCode: "11111",
  regionOrState: "state1",
  city: "city1",
  country: "country1",
  latitude: "10.99876",
  longitude: "-99.99981",
  isAdmin: false,
  isFlagged: false,
  items: [
    {
      itemID: 1,
      name: "i1",
      initialPrice: 100,
      imageURL: "test_url_1",
      condition: "great",
      description: "test item",
    },
    {
      itemID: 2,
      name: "i2",
      initialPrice: 100,
      imageURL: "test_url_2",
      condition: "great",
      description: "test item",
    },
  ],
  reviews: [],
  reports: [],
  pastPurchases: [
    {
      purchaseID: 1,
      itemName: "i4",
      sellerUser: "u3",
      imageURL: "test_url_4",
      price: 150,
      soldAt: expect.any(Date),
      fromFlaggedUser: false,
    },
  ],
};

const TEST_ITEM = {
  id: 1,
  name: "i1",
  imageURL: "test_url_1",
  initialPrice: 100,
  condition: "great",
  description: "test item",
  sellerUser: "u1",
  isSold: false,
  types: [{ id: 1, name: "electronics" }],
  location: {
    city: "city1",
    address: "100 address",
    latitude: "10.99876",
    longitude: "-99.99981",
    regionOrState: "state1",
    country: "country1",
    zipCode: "11111",
  },
};

const TEST_ITEMS = [
  {
    id: 3,
    name: "i3",
    imageURL: "test_url_3",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u2",
    types: [
      { id: 1, name: "electronics" },
      { id: 3, name: "books" },
    ],
    location: {
      city: "city2",
      regionOrState: "state2",
      country: "country2",
      zipCode: "22222",
    },
  },
  {
    id: 2,
    name: "i2",
    imageURL: "test_url_2",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [
      { id: 1, name: "electronics" },
      { id: 2, name: "movies" },
    ],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
  {
    id: 1,
    name: "i1",
    imageURL: "test_url_1",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [{ id: 1, name: "electronics" }],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
];

const TEST_MESSAGE = {
  id: 3,
  from: "u2",
  to: "u1",
  itemID: 1,
  itemName: "i3",
  body: "test body",
  createdAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGE_2 = {
  id: 3,
  from_username: "u2",
  to_username: "u1",
  itemID: 1,
  item_name: "i3",
  body: "test body",
  sent_at: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGES = {
  conversation: [
    {
      id: 2,
      from_username: "u3",
      to_username: "u1",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
    {
      id: 1,
      from_username: "u1",
      to_username: "u3",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
  ],
};

const TEST_REVIEW = {
  id: 1,
  reviewed: "u3",
  reviewer: "u1",
  rating: 10,
  body: "test review",
  madeAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_REPORT = {
  id: 1,
  reportedUser: "u2",
  reporterUser: "u1",
  body: "test body",
  madeAt: "2024-07-09 14:23:43.469126-04",
  isCleared: false,
};

const TEST_REPORTS = [
  {
    id: 1,
    reportedUser: "u2",
    reporterUser: "u1",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
  {
    id: 2,
    reportedUser: "u1",
    reporterUser: "u2",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
];

const TEST_PURCHASE = {
  purchaseID: 1,
  itemName: "i4",
  sellerUser: "u3",
  imageURL: "test_url_4",
  price: 150,
  soldAt: "2024-07-09 14:23:43.469126-04",
  fromFlaggedUser: false,
};

module.exports = {
  TEST_USER,
  TEST_ITEM,
  TEST_ITEMS,
  TEST_MESSAGE,
  TEST_MESSAGE_2,
  TEST_MESSAGES,
  TEST_REPORT,
  TEST_REPORTS,
  TEST_REVIEW,
  TEST_PURCHASE,
};
