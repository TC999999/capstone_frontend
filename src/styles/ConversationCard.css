.conversation-card {
  border: solid 2px;
  padding: 5px;
}

.conversation-card a {
  text-decoration-line: none;
}
