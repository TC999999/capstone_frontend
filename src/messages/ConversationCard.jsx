import { Link } from "react-router-dom";
import "../styles/ConversationCard.css";

const ConversationCard = ({ message }) => {
  return (
    <div className="conversation-card">
      <p>
        <b>From:</b>{" "}
        <Link to={`/users/${message.from_username}`}>
          {message.from_username}
        </Link>
      </p>
      <p>
        <b>To:</b>{" "}
        <Link to={`/users/${message.to_username}`}>{message.to_username}</Link>
      </p>
      <p>
        <b>About:</b>{" "}
        <Link to={`/items/${message.itemID}`}> {message.item_name}</Link>
      </p>
      <p>
        <b>Body:</b> {message.body}
      </p>
      <small>
        <i>sent at:</i> {message.sent_at}
      </small>
    </div>
  );
};

export default ConversationCard;
