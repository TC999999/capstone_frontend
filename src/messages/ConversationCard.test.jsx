import { render } from "@testing-library/react";
import ConversationCard from "./ConversationCard.jsx";
import { TEST_MESSAGE_2 } from "../_testCommons.js";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <ConversationCard
        message={TEST_MESSAGE_2}
        userClass={"current-user"}
        messageType={"user-messages"}
      />
    </MemoryRouter>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <ConversationCard
        message={TEST_MESSAGE_2}
        userClass={"current-user"}
        messageType={"user-messages"}
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have conversation-card-div", function () {
  const { container } = render(
    <MemoryRouter>
      <ConversationCard
        message={TEST_MESSAGE_2}
        userClass={"current-user"}
        messageType={"user-messages"}
      />
    </MemoryRouter>
  );
  const conversation_card_div = container.querySelector(
    ".conversation-card-div"
  );
  expect(conversation_card_div).toBeInTheDocument();
});

it("should have a message body", function () {
  const { getByText } = render(
    <MemoryRouter>
      <ConversationCard
        message={TEST_MESSAGE_2}
        userClass={"current-user"}
        messageType={"user-messages"}
      />
    </MemoryRouter>
  );
  const body = getByText("test body");
  expect(body).toBeInTheDocument();
});
