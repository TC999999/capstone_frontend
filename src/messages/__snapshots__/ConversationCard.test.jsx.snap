// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="conversation-card-div current-user-div"
  >
    <div
      class="conversation-card current-user user-messages"
    >
      <p>
        <b>
          From:
        </b>
         
        <a
          href="/users/u2"
        >
          u2
        </a>
      </p>
      <p>
        <b>
          To:
        </b>
         
        <a
          href="/users/u1"
        >
          u1
        </a>
      </p>
      <p>
        <b>
          About:
        </b>
         
        <a
          href="/items/1"
        >
           i3
        </a>
      </p>
      <p>
        <b>
          Body:
        </b>
         test body
      </p>
      <small>
        <i>
          sent at:
        </i>
         2024-07-09 14:23:43.469126-04
      </small>
    </div>
  </div>
</DocumentFragment>
`;
